Dockerfile

```diff
@@ -1,22 +1,20 @@
 # Dockerfile for https://github.com/adnanh/webhook
-
-FROM        alpine:3.6
-
+FROM        golang:alpine3.7 AS build
 MAINTAINER  Almir Dzinovic <almirdzin@gmail.com>
-
-ENV         GOPATH /go
-ENV         SRCPATH ${GOPATH}/src/github.com/adnanh
+WORKDIR     /go/src/github.com/adnanh/webhook
 ENV         WEBHOOK_VERSION 2.6.8
-
-RUN         apk add --update -t build-deps curl go git libc-dev gcc libgcc && \
-            curl -L -o /tmp/webhook-${WEBHOOK_VERSION}.tar.gz https://github.com/adnanh/webhook/archive/${WEBHOOK_VERSION}.tar.gz && \
-            mkdir -p ${SRCPATH} && tar -xvzf /tmp/webhook-${WEBHOOK_VERSION}.tar.gz -C ${SRCPATH} && \
-            mv -f ${SRCPATH}/webhook-* ${SRCPATH}/webhook && \
-            cd ${SRCPATH}/webhook && go get -d && go build -o /usr/local/bin/webhook && \
+RUN         apk add --update -t build-deps curl libc-dev gcc libgcc
+RUN         curl -L --silent -o webhook.tar.gz https://github.com/adnanh/webhook/archive/${WEBHOOK_VERSION}.tar.gz && \
+            tar -xzf webhook.tar.gz --strip 1 &&  \
+            go get -d && \
+            go build -o /usr/local/bin/webhook && \
             apk del --purge build-deps && \
             rm -rf /var/cache/apk/* && \
-            rm -rf ${GOPATH}
+            rm -rf /go
 
+FROM        alpine:3.7
+COPY        --from=build /usr/local/bin/webhook /usr/local/bin/webhook
+COPY        hooks.json.example /etc/webhook/hooks.json
```

Comment:
The file `hooks.json.example` does not exist in this repo, hence cannot be copied into the docker image.
Also, we actually need it, so this line can be deleted completely, we don't want to have a predefined set of hooks inside the image, we want users to define their own.